src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncTests.cs

```diff
@@ -3294,6 +3317,326 @@ .locals init (C.<f>d__0 V_0)
+            var comp = CreateCompilation(source, options: TestOptions.DebugExe);
+            comp.VerifyEmitDiagnostics(
+                // (5,22): error CS0246: The type or namespace name 'Nonexistent' could not be found (are you missing a using directive or an assembly reference?)
+                // [AsyncBuilder(typeof(Nonexistent))] class T { }
+                Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "Nonexistent").WithArguments("Nonexistent").WithLocation(5, 22)
+                );
+        }
+
+        [Fact]
+        public void AsyncTasklikeMissingBuilderType()
+        {
+            // Builder
+            var libB = @"public class B { }";
+            var cB = CreateCompilationWithMscorlib45(libB);
+            var rB = cB.EmitToImageReference();
+                
+            // Tasklike
+            var libT = @"
+using System.Runtime.CompilerServices;
+
+[AsyncBuilder(typeof(B))] public class T { }
+
+namespace System.Runtime.CompilerServices { class AsyncBuilderAttribute : System.Attribute { public AsyncBuilderAttribute(System.Type t) { } } }
+";
+            var cT = CreateCompilationWithMscorlib45(libT, references: new[] { rB });
+            var rT = cT.EmitToImageReference();
+
+            // Consumer, fails to reference builder
+            var source = @"
+using System.Threading.Tasks;
+
+class Program {
+    static void Main() { }
+    async T f() => await Task.Delay(1);
+}
+";
+            var c = CreateCompilationWithMscorlib45(source, references: new[] { rT });
+            c.VerifyEmitDiagnostics(
+                // (6,17): error CS1983: The return type of an async method must be void, Task or Task<T>
+                //     async T f() => await Task.Delay(1);
+                Diagnostic(ErrorCode.ERR_BadAsyncReturn, "=> await Task.Delay(1)").WithLocation(6, 17)
+                );
+        }
+
+
+        [Fact]
+        public void AsyncTasklikeCreateMethod()
+        {
+            var source = $@"
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+[AsyncBuilder(typeof(B0))] public class T0 {{ }}
+[AsyncBuilder(typeof(B1))] public class T1 {{ }}
+[AsyncBuilder(typeof(B2))] public class T2 {{ }}
+[AsyncBuilder(typeof(B3))] public class T3 {{ }}
+[AsyncBuilder(typeof(B4))] public class T4 {{ }}
+[AsyncBuilder(typeof(B5))] public class T5 {{ }}
+
+{AsyncBuilderCode("B0", "T0").Replace("public static B0 Create()", "public static B0 Create()")}
+{AsyncBuilderCode("B1", "T1").Replace("public static B1 Create()", "private static B1 Create()")}
+{AsyncBuilderCode("B2", "T2").Replace("public static B2 Create() => default(B2);", "public static void Create() { }")}
+{AsyncBuilderCode("B3", "T3").Replace("public static B3 Create() => default(B3);", "public static B1 Create() => default(B1);")}
+{AsyncBuilderCode("B4", "T4").Replace("public static B4 Create()", "public static B4 Create(int i)")}
+{AsyncBuilderCode("B5", "T5").Replace("public static B5 Create()", "public static B5 Create<T>()")}
```

Comment:
Consider passing the `Create()` method string to `AsyncBuilderCode()` as an optional parameter rather than using `Replace`.